packages/block-library/src/image/edit.native.js

```diff
@@ -113,10 +113,18 @@ export class ImageEdit extends Component {
 			},
 		};
 
-		this.sizeOptions = map( this.props.imageSizes, ( { name, slug } ) => ( {
-			value: slug,
-			name,
-		} ) );
+		// Only map available image sizes.
+		this.sizeOptions = map(
+			filter( this.props.imageSizes, ( { slug } ) =>
+				get( this.props.image, [
+					'media_details',
+					'sizes',
+					slug,
+					'source_url',
+				] )
+			),
```

Comment:
Hey @enejb 👋 I was testing this PR but I couldn't get the size options to show up. If I remove this filter they show up correctly. Does it work for you with the current changes?

Reply:
I was running into the same issue. 
I think it has to do if we know about the different sizes that are available. 

This filter was taken from https://github.com/WordPress/gutenberg/blob/7d87c9e20c15c5e1654b991f5882024fedc0fbc2/packages/block-library/src/image/image.js#L139

Currently, if you remove the filter and then try to set the size of the image that doesn't exist ( some images do not have all the different sizes) that change will not work as expected anyways. 

See https://github.com/WordPress/gutenberg/blob/7d87c9e20c15c5e1654b991f5882024fedc0fbc2/packages/block-library/src/image/edit.native.js#L65 

That being said. Maybe there is something mobile-specific that I don't quite understand when it comes to images and when we have the information about the different image sizes. 

cc: @mkevins Do you understand why sometimes the images do not have the different sizes available?

Reply:
> Do you understand why sometimes the images do not have the different sizes available?

I'm not entirely sure, but I imagine that it's possible from several possibilities. One could be that the available sizes are changed at some point in the site, or possibly that the uploaded image isn't converted to the various sizes from ImageMagick. Another possibility that comes to mind, if you encountered this after uploading, is that the images sizes were not updated in the component after the upload completed?

I found this comment, which might be worth investigating: https://github.com/WordPress/gutenberg/pull/28238/files#r570481358 since the sizes come from the store, but after that performance refactor are only set in the constructor. :thinking:

Reply:
As a side note, I wonder if we can remove the lodash `get` here, since optional chaining is available. Wdyt?